### PR TITLE
Working notebook integration demo

### PR DESCRIPTION
**Pull request by mgasner**

### CODE REVIEW

File: python_modules/airline-demo/airline_demo/pipelines.py [hunk 1]
```diff
@@ -138,19 +67,22 @@ def _db_load(data_frame, table_name, resources):
...
                     info.config['redshift_password'],
                     info.config['redshift_hostname'],
                     info.config['redshift_db_name'],
                 ),
-                _create_redshift_engine(
-                    info.config['redshift_username'],
-                    info.config['redshift_password'],
-                    info.config['redshift_hostname'],
-                    info.config['redshift_db_name'],
+                create_redshift_engine(
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/pipelines.py]: is the url a property of the engine? if so we could eliminate the other resource

File: python_modules/airline-demo/airline_demo/pipelines.py [hunk 2]
```diff
@@ -318,8 +256,14 @@ def define_airline_demo_download_pipeline():
...
-        ingest_csv_to_spark, join_spark_data_frames, load_data_to_database_from_spark,
-        normalize_weather_na_values, subsample_spark_dataset, thunk
+        canonicalize_column_names,
+        ingest_csv_to_spark,
+        join_spark_data_frames,
+        load_data_to_database_from_spark,
+        normalize_weather_na_values,
+        subsample_spark_dataset,
+        thunk,
+        union_spark_data_frames,
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/pipelines.py]: 👍🏻

File: python_modules/airline-demo/airline_demo/solids.py [hunk 3]
```diff
@@ -114,6 +131,20 @@ def thunk(info):
     return info.config
 
 
+@solid(
+    name='thunk_database_engine',
+    outputs=[OutputDefinition(SqlAlchemyEngineType, description='The db resource.')]
+)
+def thunk_database_engine(info):
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/solids.py]: very clever! didn't think these were serializable!

File: python_modules/airline-demo/airline_demo/pipelines.py (hunk 1 above)
Comment (mgasner) [reply python_modules/airline-demo/airline_demo/pipelines.py]: it's not a public API on `Engine`, but the real issue here is that Spark wants a jdbc URL, and SQLAlchemy doesn't

File: python_modules/airline-demo/airline_demo/solids.py (hunk 3 above)
Comment (mgasner) [reply python_modules/airline-demo/airline_demo/solids.py]: too clever by half; in fact, they are not -- i am now doing this: https://github.com/dagster-io/dagster/blob/airline-demo-sql/python_modules/airline-demo/airline_demo/pipelines.py#L389

**Pull request merged**